heuristic/heuristic_calculations.py

```diff
@@ -1,322 +1,211 @@
+             #   partial_weekend_shifts.extend([(e,t,v) for t,v in model.shifts_at_day[i] if x[e,t,v] == 1])
+              #  partial_weekend_shifts.extend([(e,t,v) for t,v in model.shifts_at_day[i+1] if x[e,t,v] == 1])
+
+            partial_weekend[e,i] =  abs((sum(x[e,t,v] 
+                                    for t,v in model.shifts_at_day[i]) 
+                                    - sum(x[e,t,v] 
+                                    for t,v in model.shifts_at_day[i+1])))
+            # if(partial_weekend[e,i] != abs(model.ro["sat"][e,i].x - model.ro["sun"][e,(i+1)].x)):
+            #     print(str(partial_weekend) + "," + str(abs(model.ro["sat"][e,i].x - model.ro["sun"][e,(i+1)].x)))
+            #     print(i)
+            #     print("Different partial weekends")
     return partial_weekend
 
-
-def calculate_isolated_working_days(data, x):
-
-    shifts_per_day = data["shifts"]["shifts_per_day"]
+def calculate_isolated_working_days(model, x):
     isolated_working_days = {}
+    for e in model.employees:
+        for i in range(len(model.days)-2):
+            isolated_working_days[e,i+1] = max(0,(-sum(x[e,t,v] for t,v in model.shifts_at_day[i]) 
+            + sum(x[e,t,v] for t,v in model.shifts_at_day[i+1]) 
+            - sum(x[e,t,v] for t,v in model.shifts_at_day[i+2])))
 
-    for e in data["staff"]["employees"]:
-        for i in range(len(data["time"]["days"]) - 2):
-            isolated_working_days[e, i + 1] = max(
-                0,
-                (
-                    -sum(x[e, t, v] for t, v in shifts_per_day[i])
-                    + sum(x[e, t, v] for t, v in shifts_per_day[i + 1])
-                    - sum(x[e, t, v] for t, v in shifts_per_day[i + 2])
-                ),
-            )
-    return isolated_working_days
+            # if(isolated_working_days[e,i] != model.q_iso["work"][e,i].x):
+            #     print("Different isolated working days")
 
+    return isolated_working_days
 
-def calculate_isolated_off_days(data, x):
 
-    shifts_per_day = data["shifts"]["shifts_per_day"]
+def calculate_isolated_off_days(model, x):
     isolated_off_days = {}
-
-    for e in data["staff"]["employees"]:
-        for i in range(len(data["time"]["days"]) - 2):
-            isolated_off_days[e, i + 1] = max(
-                0,
-                (
-                    sum(x[e, t, v] for t, v in shifts_per_day[i])
-                    - sum(x[e, t, v] for t, v in shifts_per_day[i + 1])
-                    + sum(x[e, t, v] for t, v in shifts_per_day[i + 2])
-                    - 1
-                ),
-            )
+    for e in model.employees:
+        for i in range(len(model.days)-2):
+            isolated_off_days[e,i+1] = max(0,(sum(x[e,t,v] for t,v in model.shifts_at_day[i]) 
+            - sum(x[e,t,v] for t,v in model.shifts_at_day[i+1]) 
+            + sum(x[e,t,v] for t,v in model.shifts_at_day[i+2])-1))
+        # if(isolated_off_days[e,i] != model.q_iso["off"][e,i].x):
```

Comment:
Why is this commented out?

Reply:
Test to check if it is working as it should. Should be deleted before merge.